diff --git a/config.el b/config.el
index 80fcdd3..41e286c 100644
--- a/config.el
+++ b/config.el
@@ -130,25 +130,22 @@ Return the scratch buffer opened."
 (defalias 'yes-or-no-p 'y-or-n-p)
 
 ;; leave off unless and locally set to t dependent on language.
-;; (setq-default indent-tabs-mode nil)
+(setq-default indent-tabs-mode nil)
 
 ;; make 80 the horizontal char limit
 (setq-default fill-column 80)
 
 ;; default window dimensions 130x150 and 200 pixels down. X unspecified. This
 ;; works for emacsclient as well
-(when (display-graphic-p)
-  (setq default-frame-alist
-	'((width . 130)
-	  (height . 150)
-	  (top . 200))))
+(setq default-frame-alist
+      '((width . 120)
+        (height . 100)
+        (top . 100)))
 
 ;; make firefox-developer-edition default browser
-(setq browse-url-generic-program "chromium"
+(setq browse-url-generic-program "chrome.exe"
       browse-url-browser-function #'browse-url-generic)
 
-(setq enable-recursive-minibuffers t)
-
 (add-to-list 'auto-mode-alist '("\\.service\\'" . conf-mode))
 (add-to-list 'auto-mode-alist '("\\.rules\\'" . conf-mode))
 (add-to-list 'auto-mode-alist '("\\.bashrc\\'" . shell-script-mode))
@@ -348,18 +345,15 @@ Return the scratch buffer opened."
   :config
   (add-hook 'csharp-mode-hook #'omnisharp)
   (require 'company)
-  (add-to-list 'company-backends #'company-omnisharp)
+  ;; (add-to-list 'company-backends #'company-omnisharp)
   (add-hook 'csharp-mode-hook #'flycheck-mode)
   (add-hook 'csharp-mode-hook #'eldoc-mode)
 
   (require 'csharp-mode)
   (define-key csharp-mode-map (kbd "M-.") #'omnisharp-go-to-definition)
+  (define-key csharp-mode-map (kbd "C-o") #'omnisharp-auto-complete)
   :after (company))
 
-(use-package blimp :ensure t
-  :config
-  (add-hook 'image-minor-mode-hook 'blimp-mode))
-
 ;; in buffer completion framework
 (use-package company
   :ensure t
@@ -444,9 +438,9 @@ Return the scratch buffer opened."
   ;; :ensure t
   ;; :config (add-hook 'dired-mode-hook #'dired-collapse-mode))
 
-(use-package dired-quick-sort
-  :ensure t
-  :config (dired-quick-sort-setup))     ;binds S to hyrda sort dispatcher
+;; (use-package dired-quick-sort
+  ;; :ensure t
+  ;; :config (dired-quick-sort-setup))     ;binds S to hyrda sort dispatcher
 
 (setq dired-dwim-target t)
 
@@ -583,9 +577,9 @@ Return the scratch buffer opened."
 (shell-command-to-string "man man")
 
 ;; set before requiring so as to trigger warning message when these settings don't make sense
-(setq fzf/executable "fzf")
-(setq fzf/wsl nil)			;set wsl flag
-(setq fzf/args "-x --prompt='? ' --print-query --query='!^bin !^obj '")
+(setq fzf/executable "fzf.exe")
+(setq fzf/wsl t)			;set wsl flag
+(setq fzf/args "-x --prompt='? ' --query='!^bin !^obj '")
 
 ;; forked implementation of fzf
 (require 'fzf)
diff --git a/config.org b/config.org
index 9fd3133..c0ae8fb 100644
--- a/config.org
+++ b/config.org
@@ -184,24 +184,21 @@ Update this if you change the default font face.
   (defalias 'yes-or-no-p 'y-or-n-p)
 
   ;; leave off unless and locally set to t dependent on language.
-  ;; (setq-default indent-tabs-mode nil)
+  (setq-default indent-tabs-mode nil)
 
   ;; make 80 the horizontal char limit
   (setq-default fill-column 80)
 
   ;; default window dimensions 130x150 and 200 pixels down. X unspecified. This
   ;; works for emacsclient as well
-  (when (display-graphic-p)
-    (setq default-frame-alist
-	  '((width . 130)
-	    (height . 150)
-	    (top . 200))))
+  (setq default-frame-alist
+        '((width . 120)
+          (height . 100)
+          (top . 100)))
 
   ;; make firefox-developer-edition default browser
-  (setq browse-url-generic-program "chromium"
-	browse-url-browser-function #'browse-url-generic)
-
-  (setq enable-recursive-minibuffers t)
+  (setq browse-url-generic-program "chrome.exe"
+        browse-url-browser-function #'browse-url-generic)
 #+END_SRC
 ** file mode associations
 #+BEGIN_SRC emacs-lisp
@@ -467,12 +464,13 @@ Custom keymap for c-mode. Use local-set-key to bind a prefix key from c-mode-hoo
     :config
     (add-hook 'csharp-mode-hook #'omnisharp)
     (require 'company)
-    (add-to-list 'company-backends #'company-omnisharp)
+    ;; (add-to-list 'company-backends #'company-omnisharp)
     (add-hook 'csharp-mode-hook #'flycheck-mode)
     (add-hook 'csharp-mode-hook #'eldoc-mode)
 
     (require 'csharp-mode)
     (define-key csharp-mode-map (kbd "M-.") #'omnisharp-go-to-definition)
+    (define-key csharp-mode-map (kbd "C-o") #'omnisharp-auto-complete)
     :after (company))
 #+END_SRC
 *** python                                                         :disabled:
@@ -507,8 +505,8 @@ This depends on:
     (add-to-list 'auto-mode-alist '("\\.js\\'" . js2-mode))
     (add-to-list 'interpreter-mode-alist '("node" . js2-mode)))
 #+END_SRC
-** blimp
-#+BEGIN_SRC emacs-lisp
+** blimp                                                           :disabled:
+#+BEGIN_SRC emacs-lisp :tangle no
   (use-package blimp :ensure t
     :config
     (add-hook 'image-minor-mode-hook 'blimp-mode))
@@ -608,9 +606,9 @@ Various completion functions using ivy
     ;; :ensure t
     ;; :config (add-hook 'dired-mode-hook #'dired-collapse-mode))
 
-  (use-package dired-quick-sort
-    :ensure t
-    :config (dired-quick-sort-setup))     ;binds S to hyrda sort dispatcher
+  ;; (use-package dired-quick-sort
+    ;; :ensure t
+    ;; :config (dired-quick-sort-setup))     ;binds S to hyrda sort dispatcher
 
   (setq dired-dwim-target t)
 
@@ -769,26 +767,25 @@ sub-commands
 #+END_SRC
 ** fzf
 #+BEGIN_SRC emacs-lisp
-  ;; set before requiring so as to trigger warning message when these settings don't make sense
-  (setq fzf/executable "fzf")
-  (setq fzf/wsl nil)			;set wsl flag
-  (setq fzf/args "-x --prompt='? ' --print-query --query='!^bin !^obj '")
-
-  ;; forked implementation of fzf
-  (require 'fzf)
-
-  ;; no evil mode in fzf
-  (advice-add 'fzf :after 'turn-off-evil-mode)
-  (advice-add 'fzf-git :after 'turn-off-evil-mode)
-  (advice-add 'fzf-git-grep :after 'turn-off-evil-mode)
-  (advice-add 'fzf-git-files :after 'turn-off-evil-mode)
-  (advice-add 'fzf-hg :after 'turn-off-evil-mode)
-  (advice-add 'fzf-directory :after 'turn-off-evil-mode)
-  (advice-add 'fzf-projectile :after 'turn-off-evil-mode)
-
-  (define-key semicolon-map
-    (kbd "C-f") #'fzf)
+    ;; set before requiring so as to trigger warning message when these settings don't make sense
+    (setq fzf/executable "fzf.exe")
+    (setq fzf/wsl t)			;set wsl flag
+    (setq fzf/args "-x --prompt='? ' --query='!^bin !^obj '")
+
+    ;; forked implementation of fzf
+    (require 'fzf)
+
+    ;; no evil mode in fzf
+    (advice-add 'fzf :after 'turn-off-evil-mode)
+    (advice-add 'fzf-git :after 'turn-off-evil-mode)
+    (advice-add 'fzf-git-grep :after 'turn-off-evil-mode)
+    (advice-add 'fzf-git-files :after 'turn-off-evil-mode)
+    (advice-add 'fzf-hg :after 'turn-off-evil-mode)
+    (advice-add 'fzf-directory :after 'turn-off-evil-mode)
+    (advice-add 'fzf-projectile :after 'turn-off-evil-mode)
 
+    (define-key semicolon-map
+      (kbd "C-f") #'fzf)
 #+END_SRC
 ** shell
 #+BEGIN_SRC emacs-lisp
diff --git a/init.el b/init.el
index 70bd931..78a0f11 100644
--- a/init.el
+++ b/init.el
@@ -45,27 +45,62 @@
   (org-babel-load-file (expand-file-name "~/.emacs.d/config.org")))
 ;; (load "~/.emacs.d/config")          ;byte compiling seems to have no effect on startup time
 
-
 ;;;; ===========================================================================
 ;;;;                                    desktop
 ;; (set-face-attribute 'default nil :font "fixedsys" :height 80)
-(set-face-attribute 'default nil :font "Terminus (TTF)" :height 70)
+;; (set-face-attribute 'default nil :font "Terminus (TTF)" :height 70)
 
 ;;;; ===========================================================================
 ;;;;                                     mobile
 ;; (set-face-attribute 'default nil :font "fixedsys" :height 110)
 ;; (set-face-attribute 'default nil :font "Terminus (TTF)" :height 110)
 
+;;;; ===========================================================================
+;;;;                                 windows patch
 
+;; Add these:
+;; export DISPLAY=localhost:0.0
+;; alias emacsd="emacs-gtk --daemon"
+;; alias ecl="emacsclient -c -n -d=${DISPLAY}"
 
+;; (setq-default buffer-file-coding-system 'utf-8-dos)
+;; (setq-default default-buffer-file-coding-system 'utf-8-dos)
+;; (set-default-coding-systems 'utf-8-dos)
+;; (prefer-coding-system 'utf-8-dos)
 
+(defun counsel-locate-action-extern (x)
+  "Pass X to `xdg-open' or equivalent command via the shell."
+  (interactive "FFile: ")
+  (if (and (eq system-type 'windows-nt)
+           (fboundp 'w32-shell-execute))
+      (w32-shell-execute "open" x)
+    (message (format "%s %s"
+                                        (cl-case system-type
+                                          (darwin "open")
+                                          (cygwin "cygstart")
+                                          (t "wsl-open"))
+                                        (shell-quote-argument x)))
+    (call-process-shell-command (format "%s %s"
+                                        (cl-case system-type
+                                          (darwin "open")
+                                          (cygwin "cygstart")
+                                          (t "wsl-open"))
+                                        (shell-quote-argument x))
+                                nil 0)))
 
+;; temp
+(setq default-frame-alist (append default-frame-alist
+'((inhibit-double-buffering . t))))
 
 
+(defun dired-open-in-vs ()
+  "Opens dired's current file in visual studio"
+  (interactive)
+  (let* ((filename (file-relative-name (dired-get-filename)))
+	(cmd (concat "devenv.exe " filename " /Edit")))
+    (call-process-shell-command cmd nil 0)))
 
-
-
-
+(define-key dired-mode-map (kbd "C-i") #'dired-open-in-vs)
 
 
 
@@ -105,8 +140,9 @@
  '(ibuffer-title-face (quote dired-header))
  '(package-selected-packages
    (quote
-    (omnisharp blimp gnuplot vlf htmlize not-a-package ace-window-mode dired-rsync wgrep-ag wgrep yasippet-snippets yasnippet js2-mode restclient smex web-mode evil-numbers pydoc counsel magit company-jedi slime-company company-web irony-eldoc company-irony which-key use-package tide slime rainbow-mode rainbow-delimiters pdf-tools org-bullets markdown-mode ivy helm golden-ratio evil-vimish-fold eshell-up editorconfig dired-quick-sort dired-collapse diminish company beacon ace-window)))
+    (htmlize not-a-package ace-window-mode dired-rsync wgrep-ag wgrep yasippet-snippets yasnippet js2-mode restclient smex web-mode evil-numbers pydoc counsel magit company-jedi slime-company company-web irony-eldoc company-irony which-key use-package tide slime rainbow-mode rainbow-delimiters pdf-tools org-bullets markdown-mode ivy helm golden-ratio evil-vimish-fold eshell-up editorconfig dired-quick-sort dired-collapse diminish company beacon ace-window)))
  '(pdf-view-midnight-colors (quote ("#fdf4c1" . "#1d2021")))
+ '(require-final-newline nil)
  '(vc-annotate-background "#ffffff")
  '(vc-annotate-background-mode nil)
  '(vc-annotate-color-map
